Source/JavaScriptCore/wasm/WasmAirIRGenerator.cpp

```diff
@@ -549,6 +548,19 @@ class AirIRGenerator {
         AIR_OP_CASE(ExtendHigh)
         AIR_OP_CASE(ExtendLow)
         AIR_OP_CASE(TruncSat)
+#if CPU(X86_64)
+        else if (op == SIMDLaneOperation::Neg) {
+            // x86_64 has no vector negate instruction, so we expand vxv.neg v into vxv.sub 0, v
+            // here to give B3/Air a chance to optimize out repeated vector zeroing.
+            TypedTmp zero = addConstant(v128_t());
+            result = tmpForType(Types::V128);
+            append(VectorSub, zero, v, result);
+            return { };
+        }
+#else
+        AIR_OP_CASE(Neg)
+#endif
```

Comment:
We're doing it in the AirIRGenerator in the hopes that by expanding negates as early as possible, we give the backend as many chances as possible to optimize out repeated uses of the constant. Probably not very meaningful for Air, but it may matter more in B3IRGenerator (where Air macros would expand after earlier optimization) so our plan is to keep it here so the two IRGenerators are consistent.